### PR TITLE
String handling improvements

### PR DESCRIPTION
**Pull request by akiss77**
I've reviewed some of the string handling parts of the code base. This PR show the results of my findings spearated into independent patches.

Builds on #330

### CODE REVIEW

**Comment by akiss77** [comment]
Also depends on #332

**Comment by LaszloLango** [comment]
@akiss77, please rebase and squash

**Comment by akiss77** [comment]
@LaszloLango done

File: src/iotjs_util.h [hunk 1]
```diff
@@ -51,6 +51,9 @@ class String {
   explicit String(const char* data, int size = -1, int cap = -1);
 
   // Create string object from other string object.
+  // (Actually unimplemented. Declaration of copy constructor is needed for
+  // return value optimization to work. However, linker error will be emitted
+  // if copy construction is used in any other situation.)
   String(const String& other);
```
Comment (LaszloLango) [inline src/iotjs_util.h]: IMHO this should be private if it is only for RVO.

Comment (LaszloLango) [inline src/iotjs_util.h]: I have to fix myself. That would break RVO.

**Comment by LaszloLango** [comment]
@akiss77, thanks. LGTM

**Comment by zherczeg** [comment]
LGTM

**Pull request merged**